packages/core/src/common/messaging/web-socket-channel.ts

```diff
@@ -19,7 +19,11 @@
 import { WriteBuffer } from '../message-rpc';
 import { Uint8ArrayReadBuffer, Uint8ArrayWriteBuffer } from '../message-rpc/uint8-array-message-buffer';
 import { AbstractChannel } from '../message-rpc/channel';
-import { Disposable } from '../disposable';
+import { Socket as ClientSocket } from 'socket.io-client';
+import { Socket as ServerSocket } from 'socket.io';
+import { Emitter } from 'vscode-languageserver-protocol';
+
+export type WebSocket = ClientSocket | ServerSocket;
```

Comment:
Minor: We seem declare this type multiple times. See also `socket-write-buffer.ts`